library/std/src/io/buffered/bufwriter.rs

```diff
@@ -385,3 +555,22 @@ impl<W: Write> Drop for BufWriter<W> {
         }
     }
 }
+
+/// Similar to iter.find, this method searches an iterator for an item
+/// matching a predicate, but returns it only if it is the *only* item
+/// matching that predicate. Used to check if there is exactly one non-empty
+/// buffer in a list input to write_vectored.
+///
+/// FIXME: delete this function and replace it with slice::trim if that becomes
+/// a things (https://github.com/rust-lang/rfcs/issues/2547)
+fn only_one<I, T>(iter: I, filter: impl FnMut(&T) -> bool) -> Option<T>
+where
+    I: IntoIterator<Item = T>,
+    I::IntoIter: FusedIterator,
+{
+    let mut iter = iter.into_iter().filter(filter);
+    match (iter.next(), iter.count()) {
+        (Some(item), 0) => Some(item),
+        _ => None,
+    }
```

Comment:
This can be short-circuited:

```rust
    match (iter.next(), iter.next()) {
        (Some(item), None) => Some(item),
        _ => None,
    }
```

Reply:
Oh, good idea

Reply:
Done